torch/distributed/pipeline/sync/pipe.py

```diff
@@ -137,64 +137,65 @@ def _split_module(modules: nn.Sequential) -> Tuple[List[nn.Sequential], List[tor
 
-    You should place all the modules on the appropriate devices before passing
-    them to this API and wrap them into an ``nn.Sequential`` module defining the
+    You should place all the modules on the appropriate devices and wrap them into an :class:`nn.Sequential <torch.nn.Sequential>` module defining the
     desired order of execution.
 
-    Args:
-        module (``torch.nn.Sequential``):
-            Sequential module to be parallelized using pipelining. Each module
+    Arguments:
+        module (:class:`nn.Sequential <torch.nn.Sequential>`):
+            sequential module to be parallelized using pipelining. Each module
             in the sequence has to have all of its parameters on a single
             device. Each module in the sequence has to either be an nn.Module
-            or ``nn.Sequential`` (to combine multiple sequential modules on a single
+            or :class:`nn.Sequential <torch.nn.Sequential>` (to combine multiple sequential modules on a single
             device)
-
-    Keyword Args:
         chunks (int):
             number of micro-batches (default: ``1``)
         checkpoint (str):
             when to enable checkpointing, one of ``'always'``,
-            ``'except_last'``, or ``'never'`` (default: ``'except_last'``)
+            ``'except_last'``, or ``'never'`` (default: ``'except_last'``). ``'never'`` disables checkpointing completely, ``'except_last'`` enables checkpointing for all micro-batches except the last one and ``'always'`` enables checkpointing for all micro-batches.
         deferred_batch_norm (bool):
             whether to use deferred BatchNorm moving statistics (default:
-            :data:`False`, see :ref:`Deferred Batch Normalization` for more
-            details)
+            :data:`False`). If set to :data:`True`, we track statistics across multiple micro-batches to update the running statistics per mini-batch.
 
     Raises:
         TypeError:
             the module is not a :class:`nn.Sequential <torch.nn.Sequential>`.
         ValueError:
             invalid arguments
-        IndexError:
-            the number of devices is fewer than the number of partitions.
 
-    """
+    Example::
+        Pipeline of two FC layers across GPUs 0 and 1.
 
-    #: The number of micro-batches.
-    chunks: int = 1
+        >>> fc1 = nn.Linear(16, 8).cuda(0)
+        >>> fc2 = nn.Linear(8, 4).cuda(1)
+        >>> model = nn.Sequential(fc1, fc2)
+        >>> model = Pipe(model, chunks=8)
+        >>> input = torch.rand(16, 16).cuda(0)
+        >>> output_rref = model(input)
 
-    #: The checkpoint mode to determine when to enable checkpointing. It is one
-    #: of ``'always'``, ``'except_last'``, or ``'never'``.
-    checkpoint: str = "except_last"
+    .. note::
+        You can wrap a :class:`Pipe` model with
+        :class:`torch.nn.parallel.DistributedDataParallel` only when the
+        checkpoint parameter of :class:`Pipe` is ``'never'``.
```

Comment:
are we going to check this?

Reply:
No we are not checking this, this is just a note to the user that other modes will not work with DDP.